plugins/rich-editor/src/scripts/Editor/Generic/MenuItem.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * @author Adam (charrondev) Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+import React from "react";
+import * as PropTypes from "prop-types";
+import classnames from "classnames";
+import { t } from "@core/application";
+import * as Icons from "../Icons";
+
+type FormatterCallback = (IMenuItemData) => void;
+
+export interface IMenuItemData {
+    active: boolean;
+    label?: string;
+    formatName?: string;
+    formatter?: FormatterCallback;
+    enableValue?: object;
+}
+
+interface IProps {
+    propertyName: string;
+    label: string;
+    isActive: boolean;
+    isFirst: boolean;
+    isLast: boolean;
+    onClick: (event: React.MouseEvent<any>) => void;
+    onBlur: (event?: React.FocusEvent<any>) => void;
+    role?: string;
+}
+
+/**
+ * Component for a single item in a EditorToolbar.
+ */
+export default class MenuItem extends React.Component<IProps> {
+
+    private onBlur: (event?: React.FocusEvent<any>) => void;
+    private buttonRole: string;
+    private domButton: HTMLElement;
+
+    constructor(props) {
+        super(props);
+        this.onBlur = (props.isLast && props.onBlur) ? props.onBlur : () => undefined;
+        this.buttonRole = props.role || "button";
+    }
+
+    public render() {
+        const { propertyName, isActive, onClick } = this.props;
+        const Icon = Icons[propertyName];
+        const buttonClasses = classnames("richEditor-button", "richEditor-menuItem", {
+            isActive: isActive || false,
+        });
+
+        return <button
+            ref={(ref: HTMLButtonElement) => { this.domButton = ref; }}
+            className={buttonClasses}
+            type="button"
+            aria-label={t('richEditor.menu.' + this.props.propertyName)}
```

Comment:
The translation isn't coming through. 
<img width="272" alt="screen shot 2018-04-06 at 13 46 34" src="https://user-images.githubusercontent.com/19651440/38435854-fd92e610-39a0-11e8-85e7-c68891182df2.png">

Reply:
I've updated these translations